language-design.md

```diff
@@ -17,7 +17,11 @@ For each quote, describe what it was about the quote that led you pick it.
 
+However, while designing the perfect programming language may be an impossible task, there are some good design principles that any successful programming language should follow. Joshua Bloch, writing specifically about API design writes: “**Obey the principle of least astonishment.** Every method should do the least surprising thing it could, given its name. If a method doesn’t do what users think it will, bugs will result.” [Bloch, 2006]. Most programmers’ immediate reaction to this quote is a small laugh and a murmured “Ugh, yes”. A function called max() must return the maximum number out of some list of numbers, if it doesn’t, users will assume it does, use it that way, and then spend hours searching their code for the bug. If they find it, they will only be more frustrated that a fault in the language made them waste so much time. Programming languages may seem cryptic to novices, but they follow general rules and guidelines that should be true across all languages even if they vary in the exact way they implement things. Programmers will not put up with languages that break these rules in too many ways.
+
+John Pavlus argues: “The broad computer science academic community has not paid a tremendous amount of attention to programming language usability” [Pavlus, 2012]. We absolutely disagree. Programming languages are first and foremost, languages. To an outsider who does not know the language, it may look incomprehensible, in the same way that Spanish may look incomprehensible to someone who only knows English. However, it has meaning, it follows general rules. When programming languages are written, they are written to be used by programmers. Creators do think about usability. They attempt to follow rules like “**The principle of least astonishment**” but they make tradeoffs between usability and other goals. Usability is not the only thing to consider when writing a language, but that doesn’t mean that programmers do not care about it at all.
 
```

Comment:
The key word here is "academic". What Pavlus is saying is that there haven't been that many papers / studies about language design, relative to the number of papers about language _implementation_. It's true that most language creators think about their users. But many (most?) popular language creators aren't academics and don't write papers about their languages. As a specific example, in the 1990's, people added generic types to Java (e.g., List[Int]). PL researchers worked out how to do so in [some papers](http://homepages.inf.ed.ac.uk/wadler/gj/Documents/), but none of those papers directly studied how the changes would affect programmers (e.g., would it be easy for programmers to use them?).